extensions/azurePublishNew/src/node/availableResources.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { ProvisionConfig, ResourceConfig, ResourceDefinition, ResourceProvisionService } from './types';
+import { appInsightsDefinition, getAppInsightsProvisionService } from './azureResources/appInsights';
+import { appRegistrationDefinition, getAppRegistrationProvisionService } from './azureResources/appRegistration';
+import { azureFunctionDefinition, getAzureFunctionsProvisionService } from './azureResources/azureFunction';
+import { blobStorageDefinition, getBlogStorageProvisionService } from './azureResources/blobStorage';
+import { botRegistrationDefinition, getBotChannelProvisionService } from './azureResources/botChannel';
+import { cosmosDbDefinition, getCosmosDbProvisionService } from './azureResources/cosmosDb';
+import { getLuisAuthoringProvisionService, luisAuthoringDefinition } from './azureResources/luisAuthoring';
+import { getLuisPredictionProvisionService, luisPredictionDefinition } from './azureResources/luisPrediction';
+import { getQnAProvisionService, qnaDefinition } from './azureResources/qna';
+import { getAppServiceProvisionService, servicePlanDefinition } from './azureResources/servicePlan';
+import { getWebAppProvisionService, webAppResourceDefinition } from './azureResources/webApp';
+
+export const availableResources: ResourceDefinition[] = [
```

Comment:
I put the available resources and the creation of the services together because if you register one you want to register the other.